### PR TITLE
MOSFIRE flatlampoff

### PR DESCRIPTION
**Pull request by deborape**
PypeIt file updated for K band observation to include the `thermalflat` frame type

### CODE REVIEW

**Review by jhennawi — APPROVED** [review]
Let's discuss the name of the thermal flat and also please don't merge until objfind_refactor clears.

File: pypeit_files/keck_mosfire_mask1_k_with_continuum.pypeit [hunk 1]
```diff
@@ -22,9 +22,9 @@ data read
...
-#| m121128_0105.fits |           pixelflat,trace |         7.8 |        45.0 |       unknown | K-spectroscopy | ic348_TK_M03A |     1,1 | 56259.14200914 | 1.41291034 |  14.5479 |       K |        off |    none |    none |     0.0 |     105 |   all |      -1 |     -1 |
-#| m121128_0106.fits |           pixelflat,trace |         7.8 |        45.0 |       unknown | K-spectroscopy | ic348_TK_M03A |     1,1 | 56259.14231181 | 1.41291034 |  14.5479 |       K |        off |    none |    none |     0.0 |     106 |   all |      -1 |     -1 |
-#| m121128_0107.fits |           pixelflat,trace |         7.8 |        45.0 |       unknown | K-spectroscopy | ic348_TK_M03A |     1,1 | 56259.14262084 | 1.41291034 |  14.5479 |       K |        off |    none |    none |     0.0 |     107 |   all |      -1 |     -1 |
+| m121128_0105.fits |               thermalflat |         7.8 |        45.0 |       unknown | K-spectroscopy | ic348_TK_M03A |     1,1 | 56259.14200914 | 1.41291034 |  14.5479 |       K |        off |    none |    none |     0.0 |     105 |   all |      -1 |     -1 |
```
Comment (jhennawi) [inline pypeit_files/keck_mosfire_mask1_k_with_continuum.pypeit:25]: See my comment on the naming on the main PR, i.e. I advocated lampoffflat.

**Review by jhennawi — CHANGES_REQUESTED** [review]
Hi @deborape, 

Can you please also fix the keck_mosfire_y_long.pypeit pypeit file to now use this lampoffflat system. It is currently using a hack using bias frames. I think that is the only other mosfire dev suite pypeit file that uses lampoffflats, but it is worth it to check. See a separate comment in the PypeIt Developer Slack on the J1 filter, which appears to currently not yet be tested.

File: pypeit_files/keck_mosfire_mask1_k_with_continuum.pypeit (hunk 1 above)
Comment (deborape) [reply pypeit_files/keck_mosfire_mask1_k_with_continuum.pypeit]: changed to `lampoffflats`

File: pypeit_files/keck_mosfire_j2_long.pypeit [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+| m220409_0019.fits |                     trace |        230.0 |        45.0 |     DOME FLATS | J2-spectroscopy | LONGSLIT-46x1 |     1,1 | 59678.24498093 | 1.41291034 |   8.72874 |      J2 |         on |    Stare |  object |     0.0 |      19 |   all |      -1 |     -1 |
+| m220409_0020.fits |                     trace |        230.0 |        45.0 |     DOME FLATS | J2-spectroscopy | LONGSLIT-46x1 |     1,1 | 59678.24518753 | 1.41291034 |   8.72874 |      J2 |         on |    Stare |  object |     0.0 |      20 |   all |      -1 |     -1 |
+| m220409_0021.fits |                     trace |        230.0 |        45.0 |     DOME FLATS | J2-spectroscopy | LONGSLIT-46x1 |     1,1 | 59678.24540003 | 1.41291034 |   8.72874 |      J2 |         on |    Stare |  object |     0.0 |      21 |   all |      -1 |     -1 |
+| m220409_0022.fits |                      dark |        230.0 |        45.0 |     DOME FLATS | J2-spectroscopy | LONGSLIT-46x1 |     1,1 | 59678.24685211 | 1.41291034 |   8.72874 |      J2 |         on |    Stare |  object |     0.0 |      22 |   all |      -1 |     -1 |
```
Comment (jhennawi) [inline pypeit_files/keck_mosfire_j2_long.pypeit:43]: This should be lampoffflat not dark right? Also, does your lampofflat functionality work with the trace frames. It absolutely should and if it does not we need to implement that.

**Review by jhennawi — CHANGES_REQUESTED** [review]
@deborape thanks for this. The following set of .pypeit files also needs to be updated to use the lampoffflats functionality: 

p200_tspec_tspec.pypeit
vlt_sinfoni_k_0.8.pypeit
vlt_xshooter_nir.pypeit
vlt_xshooter_nir_TEST.pypeit

I'm not sure whether vlt_xshooter_nir_TEST.pypeit is actually used (I think not), but the others for sure. 

There are some hacks in those PypeIt files using baseprocess or process keywords as you will see that we were using to get this functionality. That is surely very confusing to people and should all be unnecessary now given your improvements. It would be great if you can also clean those up. 

Thanks for doing this!

**Comment by deborape** [comment]
@jhennawi the lampoffflat were not being subtracted from the trace frames, but I added it now. See PR https://github.com/pypeit/PypeIt/pull/1410

I did cleaned up the other pypeit files you suggested. I'll re-run the Dev Suite

File: pypeit_files/ldt_deveny_dv9.pypeit [hunk 3]
```diff
@@ -20,7 +20,7 @@ data read
 |            filename |       frametype |                 ra |                dec |        target |        dispname | decker | binning |                mjd | airmass | exptime | cenwave | filter1 | dispangle | slitwid | lampstat01 |
 |  20211114.0045.fits |        arc,tilt |  68.09841666666665 | 17.751472222222223 |    Cd Hg Lamp | DV9 (1200/5000) |   None |     1,1 |  59532.18842210648 |    1.71 |    30.0 |  4495.0 |   CLEAR |     35.21 |     2.5 |     Cd, Hg |
 |  20211114.0001.fits |            bias | 290.31374999999997 |  34.75936111111111 |          Bias | DV9 (1200/5000) |   None |     1,1 |  59531.98357164352 |     1.0 |     0.0 |  4000.0 |   CLEAR |      33.2 |     1.0 |        off |
-| 20211114.0032b.fits |            dark | 290.31374999999997 |  34.75936111111111 | Dome Flat OFf | DV9 (1200/5000) |   None |     1,1 |  59532.01837893519 |    1.47 |    60.0 |  4495.0 |   CLEAR |     35.21 |     1.0 |        off |
```
Comment (jhennawi) [inline pypeit_files/ldt_deveny_dv9.pypeit:23]: Good catch. Is this an IR spectrograph?

**Comment by jhennawi** [comment]
Nice work @deborape. These pypeit files are much simpler to understand now.

**Review by jhennawi — APPROVED** [review]

**Comment by deborape** [comment]
merging

**Pull request merged**